api/src/main/java/org/openmrs/api/db/hibernate/HibernateProviderDAO.java

```diff
@@ -99,26 +100,29 @@ public Provider getProviderByUuid(String uuid) {
 	 */
 	@Override
 	public Collection<Provider> getProvidersByPerson(Person person, boolean includeRetired) {
-		Criteria criteria = getSession().createCriteria(Provider.class);
+		Session session = sessionFactory.getCurrentSession();
+		CriteriaBuilder cb = session.getCriteriaBuilder();
+		CriteriaQuery<Provider> cq = cb.createQuery(Provider.class);
+		Root<Provider> root = cq.from(Provider.class);
+
+		Predicate personPredicate = cb.equal(root.get("person"), person);
 		if (!includeRetired) {
-			criteria.add(Restrictions.eq("retired", false));
+			cq.where(personPredicate, cb.isFalse(root.get("retired")))
```

Comment:
Wasn't the order by clause outside of the `if (!includeRetired)`?

Reply:
Yea, but the logic should be the same, I just restructured it a little to avoid new lists. I updated the code though to make it closer to the original structure